jerry-core/parser/js/js-parser-expr.c

```diff
@@ -607,6 +612,12 @@ parser_parse_class (parser_context_t *context_p, /**< context */
 
   JERRY_ASSERT (context_p->token.type == LEXER_RIGHT_BRACE);
 
+  parser_emit_cbc_literal (context_p, CBC_PUSH_LITERAL, class_ident_index);
```

Comment:
Good catch! I think we can use the new code block like that:

```c
if (!is_class_env)  {
// code here
}
```

Since if the inheritance is present, the constructor set [here](https://github.com/pando-project/jerryscript/blob/master/jerry-core/vm/vm.c#L1398).

Reply:
After several tries, I found that changing byte codes might not be the elegant way to implement the feature. I've update the PR, could you have another review? :D